messages.go

```diff
@@ -8,9 +8,56 @@ package notmuch
 // #include <stdlib.h>
 // #include <notmuch.h>
 import "C"
+import "runtime"
 
 // Messages represents notmuch messages.
 type Messages struct {
 	cptr   *C.notmuch_messages_t
 	thread *Thread
 }
+
+// Next retrieves the next message from the result set. Next returns true if a message
+// was successfully retrieved.
+func (ms *Messages) Next(m *Message) bool {
+	if !ms.valid() {
+		return false
+	}
+	*m = *ms.get()
+	C.notmuch_messages_move_to_next(ms.cptr)
+	return true
+}
+
+// Tags return a list of tags from all messages.
+//
+// WARNING: You can no longer iterate over messages after calling this
+// function, because the iterator will point at the end of the list.  We do not
+// have a function to reset the iterator yet and the only way how you can
+// iterate over the list again is to recreate the message list.
```

Comment:
The docs in notmuch.h says this returns NULL on error. We should handle that;
right now we'll end up returning a *Tags with a nil cptr. Unfortunately it
doesn't say what sorts of errors can occur and there's no _st variant. Maybe
it's just OOM? we should pin that down.